Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -65,4 +65,25 @@ public record DafnyDocument(
 
     public int LinesCount => VerificationTree.Range.End.Line;
   }
+
+  public record DocumentTextBuffer : TextDocumentItem {
+    public static DocumentTextBuffer From(TextDocumentItem textDocumentItem) {
+      return new DocumentTextBuffer {
+        Text = textDocumentItem.Text,
+        Uri = textDocumentItem.Uri,
+        NumberOfLines = ComputeNumberOfLines(textDocumentItem.Text),
+        Version = textDocumentItem.Version,
+        LanguageId = textDocumentItem.LanguageId
+      };
+    }
+
+    public int NumberOfLines { get; init; } //
+
+    public static int ComputeNumberOfLines(string text) {
```

Comment:
I would put this function and the next in TextChangeProcessor and make `NumberOfLines` a parameter of `DocumentTextBuffer`